File: computer-simulation-practice-code/ram.cpp
#include <iostream>
#include <string>
using namespace std;

class ram {

private:
    int* capacity;

public:
    ram(int size) {
        capacity = new int[size];
}
    
void write(int a, int b) {
    capacity[a] = b;
}

void read(int a){
    cout << "I am ram          ";
    cout << capacity[a];
}

void initialize(int adress, string a){
//starting from int adress, fill the array with each character from a

    for(int i = 0; i < a.length(); i++){
        char myChar = a[i];
        int a = int(myChar);
        capacity[adress] = myChar;
        adress++;
        }
}

};

File: JAVA/ram.java

public class ram implements memory {

    private int size;
    private int[] capacity;

    public ram(int a) {
        size = a;
        capacity = new int[size];

    }

    @Override
    public void initialize(int address, String word) {

        System.out.println("RAM initialize ");
        for (int i = 0; i < word.length(); i++) {
            int a = (int) word.charAt(i);

            capacity[address] = a;
            address++;
        }
    }

    public int getSize() {
        return size;
    }

    @Override
    public int read(int address) {
        int toRead = capacity[address];
        return toRead;
    }

    @Override
    public void write(int address, int a) {
        capacity[address] = a;
    }
}